The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10201440481/job/28223289163) [(plain)](https://github.com/rust-lang/rust/commit/b3c65835fb7568d78985a8444f384f4cc02f4e1b/checks/28223289163/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:3aacb9c90579defe09351ac5e8ee504359f8054da6326ff19038f1b7c90e3cb2aafe33685c6d9b76ee8d2ccbd187ca80c46ab5380485abdd8c0ce7d69cd8d8fd:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
failures:

---- [codegen] tests/codegen/ptr-discriminant-update.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-17/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-discriminant-update/ptr-discriminant-update.ll" "/checkout/tests/codegen/ptr-discriminant-update.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
--- stderr -------------------------------
/checkout/tests/codegen/ptr-discriminant-update.rs:17:17: error: CHECK-NEXT: is not on the line after the previous match
 // CHECK-NEXT: store i8
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-discriminant-update/ptr-discriminant-update.ll:16:2: note: 'next' match was here
 store i8 1, ptr %s, align 1
 ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-discriminant-update/ptr-discriminant-update.ll:8:6: note: previous match ended here
     ^
     ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-discriminant-update/ptr-discriminant-update.ll:9:1: note: non-matching line after previous match is here
 %_5.sroa.2 = alloca [753 x i8], align 1

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/ptr-discriminant-update/ptr-discriminant-update.ll
Check file: /checkout/tests/codegen/ptr-discriminant-update.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
<<<<<<
         1: ; ModuleID = 'ptr_discriminant_update.87432f906ccb1baa-cgu.0' 
         2: source_filename = "ptr_discriminant_update.87432f906ccb1baa-cgu.0" 
         3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128" 
         4: target triple = "x86_64-unknown-linux-gnu" 
         5:  
         6: ; Function Attrs: mustprogress nofree nosync nounwind nonlazybind willreturn memory(argmem: readwrite) uwtable 
         7: define void @update(ptr nocapture noundef %s) unnamed_addr #0 { 
         8: start: 
         9:  %_5.sroa.2 = alloca [753 x i8], align 1 
        10:  %_3 = alloca [754 x i8], align 1 
        11:  call void @llvm.lifetime.start.p0(i64 754, ptr nonnull %_3) 
        12:  call void @llvm.memcpy.p0.p0.i64(ptr noundef nonnull align 1 dereferenceable(754) %_3, ptr noundef nonnull align 1 dereferenceable(754) %s, i64 754, i1 false) 
        13:  %0 = getelementptr inbounds i8, ptr %_3, i64 1 
        14:  call void @llvm.memcpy.p0.p0.i64(ptr noundef nonnull align 1 dereferenceable(753) %_5.sroa.2, ptr noundef nonnull align 1 dereferenceable(753) %0, i64 753, i1 false) 
        15:  call void @llvm.lifetime.end.p0(i64 754, ptr nonnull %_3) 
        16:  store i8 1, ptr %s, align 1 
next:17      !~~~~~~~                     error: match on wrong line
        17:  %_5.sroa.2.0.s.sroa_idx = getelementptr inbounds i8, ptr %s, i64 1 
        18:  call void @llvm.memcpy.p0.p0.i64(ptr noundef nonnull align 1 dereferenceable(753) %_5.sroa.2.0.s.sroa_idx, ptr noundef nonnull align 1 dereferenceable(753) %_5.sroa.2, i64 753, i1 false) 
        19:  ret void 
        20: } 
        21:  
        22: ; Function Attrs: mustprogress nocallback nofree nounwind willreturn memory(argmem: readwrite) 
        23: declare void @llvm.memcpy.p0.p0.i64(ptr noalias nocapture writeonly, ptr noalias nocapture readonly, i64, i1 immarg) #1 
        24:  
        25: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
        26: declare void @llvm.lifetime.start.p0(i64 immarg, ptr nocapture) #2 
        27:  
        28: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
        29: declare void @llvm.lifetime.end.p0(i64 immarg, ptr nocapture) #2 
        30:  
        31: attributes #0 = { mustprogress nofree nosync nounwind nonlazybind willreturn memory(argmem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
        32: attributes #1 = { mustprogress nocallback nofree nounwind willreturn memory(argmem: readwrite) } 
        33: attributes #2 = { mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) } 
        34:  
        35: !llvm.module.flags = !{!0, !1} 
        36: !llvm.ident = !{!2} 
        37:  
        38: !0 = !{i32 8, !"PIC Level", i32 2} 
        39: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
        40: !2 = !{!"rustc version 1.82.0-nightly (7cae25bf4 2024-08-01)"} 
------------------------------------------



```

</details>

--- a/tests/codegen/issues/issue-122600-ptr-discriminant-update.rs
+++ b/tests/codegen/issues/issue-122600-ptr-discriminant-update.rs
@@ -1,0 +1,19 @@
+//@ compile-flags: -O
+//@ min-llvm-version: 19
+
+#![crate_type = "lib"]
+
+pub enum State {
+    A([u8; 753]),
+    B([u8; 753]),
+}
+
+// CHECK-LABEL: @update
+#[no_mangle]
+pub unsafe fn update(s: *mut State) {
+    // CHECK-NEXT: start:
+    // CHECK-NEXT: store i8
+    // CHECK-NEXT: ret
+    let State::A(v) = s.read() else { std::hint::unreachable_unchecked() };
+    s.write(State::B(v));
+}

